Modules/Core/Common/include/itkMacro.h

```diff
@@ -1218,14 +1218,11 @@ compilers.
   kernel                                                                                                               \
   {                                                                                                                    \
   public:                                                                                                              \
+    ITK_DISALLOW_COPY_AND_MOVE(kernel);                                                                                \
+    kernel() = delete;                                                                                                 \
+    ~kernel() = delete;                                                                                                \
```

Comment:
delete on a non-virtual destructor? My alarms just switched on, can you expand on this? Where have you seen this pattern? Did you mean default?

Reply:
if this class cannot be constructed, and not deleted, is there any better way to represent it?

Reply:
Ok, I see this limits the usage to only the static function.